code/modules/atmospherics/gasmixtures/gas_mixture.dm

```diff
@@ -128,10 +135,18 @@ GLOBAL_LIST_INIT(gaslist_cache, init_gaslist_cache())
 
 	return 1
 
-	///Merges all air from giver into self. Deletes giver. Returns: 1 if we are mutable, 0 otherwise
+	///Merges all air and volume from giver into self. Returns: 1 if we are mutable, 0 otherwise
+/datum/gas_mixture/proc/full_merge(datum/gas_mixture/giver)
+	if(!merge(giver))
+		return 0
+	volume += giver.volume
+	return 1
+
 /datum/gas_mixture/proc/merge(datum/gas_mixture/giver)
 	if(!giver)
 		return 0
+	if(src == giver)//self merging is sin!
+		to_chat(world, "## TESTING: self merging is sin!")

```

Comment:
is there a reason for this?

Reply:
i remove `if(src == giver)` check  from places where `merge()` called. and need see that i dont mismach some logick and nothing called self merging. 
Selfmerging just increase gas amount twice.